Issue #11 Fixed

BraxtonElmer opened the pull request:
Fixes issue #11 
The code in Digital_guj_scraper.py now web scrapes the link and sees the notifications, and sends a mail to users and saves all of the current notifications in the notifications.json file, and when it is run again, the data is compared with notifications.json and if there is new notifications in the web, it is again mailed to the user and then the new notifications are saved in the notifications.json file.

BraxtonElmer committed:
Web Scraping

BraxtonElmer committed:
Fixed text not having links being added to format

BraxtonElmer committed:
added Json feature to save notifs

BraxtonElmer committed:
added email support for new notification.

BraxtonElmer committed:
added the __name__ == '__main__':

Rjchauhan18 commented on Digital_guj_scraper.py:
```diff
@@ -63,9 +31,7 @@ def Sendemail(date,receivers_email):
-    # html = (date + '\n'+Notification+ '\n'+link)
-    html = (date )
-
+    html = (date + '\n'+Notification+ '\n'+link)
```
Fix this 

![image](https://github.com/Rjchauhan18/gtu_notification/assets/113838908/d7a40174-0d8c-411c-8e39-af98653bfa86)

BraxtonElmer committed:
Fixed issues with email and better readability

BraxtonElmer commented:
@Rjchauhan18 The error you mentioned has been fixed in my latest commit. I have also done the required testing to ensure that the code functions properly.

Rjchauhan18 commented on Digital_guj_scraper.py:
```diff
@@ -13,12 +13,11 @@
 
 # smtp connection
-def Sendemail(date,Notification, link,receivers_email):
+def Sendemail(date,Notification, link):
```
Can you please improve mail text by removing this unwanted texts.

![image](https://github.com/Rjchauhan18/gtu_notification/assets/113838908/7c52297c-f0f9-4162-b244-581e1147e163)

BraxtonElmer replied:
hello, can you please explain what you mean by unnecessary text in this image? All of them are present in the notification part of the webpage

Rjchauhan18 commented on Digital_guj_scraper.py:
```diff
@@ -13,12 +13,11 @@
+def Sendemail(date,Notification, link):
   # Define email addresses to use
-    sender_email = os.environ.get('SMTP_SENDER_EMAIL')#sender email
+    sender_email =os.environ.get('SMTP_SENDER_EMAIL')#sender email
```
please import dotenv module for getting this variables from .env file

Rjchauhan18 commented:
Your are going in the right direction just made these small changes .

Rjchauhan18 commented on Digital_guj_scraper.py:
```diff
@@ -1,56 +1,23 @@
 from bs4 import BeautifulSoup
+import os
+from os import path
+import pickle
```
remove the modules which are not use and also update requirements.txt if any new module is added.

BraxtonElmer committed:
fixed env variables loading

BraxtonElmer committed:
fixed env variable names in readme

BraxtonElmer commented:
I have added dotenv library and fixed env variables loading now, and added dotenv to requirements. I have also fixed the typo in env variable name in readme and added a variable for receiver email.
